clap_man/src/render.rs

```diff
@@ -0,0 +1,197 @@
+            (Some(short), Some(long)) => format!("{}-{}|--{}{} ", lhs, short, long, rhs),
+            (Some(short), None) => format!("{}-{}{} ", lhs, short, rhs),
+            (None, Some(long)) => format!("{}--{}{} ", lhs, long, rhs),
+            (None, None) => "".to_string(),
+        });
+    }
+
+    for arg in app.get_positionals() {
+        let (lhs, rhs) = optional_or_required_markers(arg);
+        res.push_str(&format!("{}{}{} ", lhs, arg.get_name(), rhs));
+    }
+
+    if app.has_subcommands() {
+        res.push_str(&format!("[{}] ", subcommand_heading(app).to_lowercase()));
+    }
+
+    res
+}
+
+pub(crate) fn options(app: &clap::App) -> Vec<String> {
+    let mut res = Vec::new();
+    let items: Vec<_> = app
+        .get_arguments()
+        .filter(|i| !i.is_set(ArgSettings::Hidden))
+        .collect();
+
+    for opt in items.iter().filter(|a| !a.is_positional()) {
+        let mut body = Vec::new();
+
+        let mut header = match (opt.get_short(), opt.get_long()) {
+            (Some(short), Some(long)) => {
+                vec![short_option(short), ", ".to_string(), long_option(long)]
+            }
+            (Some(short), None) => vec![short_option(short)],
+            (None, Some(long)) => vec![long_option(long)],
+            (None, None) => vec![],
+        };
+
+        if let Some(value) = &opt.get_value_names() {
+            header.push(format!("={}", italic(&value.join(" "))));
+        }
+
+        if let Some(defs) = option_default_values(opt) {
+            header.push(format!(" {}", defs));
+        }
+
+        if let Some(help) = opt.get_long_help().or_else(|| opt.get_help()) {
+            body.push(help.to_string());
+        }
+
+        if let Some(env) = option_environment(opt) {
+            body.push(env);
+        }
+
+        body.push("\n".to_string());
+
+        res.push(list(&header, &body));
+    }
+
+    for pos in items.iter().filter(|a| a.is_positional()) {
+        let (lhs, rhs) = optional_or_required_markers(pos);
+        let name = format!("{}{}{}", lhs, pos.get_name(), rhs);
+
+        let mut header = vec![bold(&name)];
+
+        let mut body = Vec::new();
+
+        if let Some(defs) = option_default_values(pos) {
+            header.push(format!(" {}", defs));
+        }
+
+        if let Some(help) = pos.get_long_help().or_else(|| pos.get_help()) {
+            body.push(help.to_string());
+        }
+
+        if let Some(env) = option_environment(pos) {
+            body.push(env);
+        }
+
+        res.push(list(&header, &body))
+    }
+
+    res
+}
+
+pub(crate) fn subcommands(app: &clap::App, section: i8) -> Vec<String> {
+    app.get_subcommands()
+        .filter(|s| !s.is_set(AppSettings::Hidden))
+        .map(|command| {
+            let name = format!("{}-{}({})", app.get_name(), command.get_name(), section);
+
+            let mut body = match command.get_long_about().or_else(|| command.get_about()) {
```

Comment:
Like with `--help`, we should probably prefer `get_about` over `get_long_about` since that is what the subcommand man pages are for.

(just noticed `--help`s precedence with a recent bug report on it)

Reply:
Fixed.